File: platform/android/library/src/main/java/com/example/module_rtsp/HelloPlayError.java
package com.example.module_rtsp;

/**
 * create by liaohailong
 * 2025/2/9 9:39
 * desc: 对应 HelloPlayerError.hpp
 */
public enum HelloPlayError {
    UnKnown(-1), // 未知错误，底层上报的错误，应用上层对应不上

    None(0), // 无错误，表示成功

    NoDataSource(-100), // 资源没有设置
    NoStreamInDataSource(-101), // 资源中没有任何轨道信息
    NoAudioVideoStream(-102), // 资源中没有音视频轨道

    NetworkTimeout(-400), // 网络连接超时


    ;

    public final int code;

    HelloPlayError(int code) {
        this.code = code;
    }
}

File: platform/ohos/helloplayer/src/main/ets/player/HelloPlayError.ts
// 定义 HelloPlayError 枚举
export const enum HelloPlayError {
  UnKnown = -1,
  None = 0,
  NoDataSource = -100,
  NoStreamInDataSource = -101,
  NoAudioVideoStream = -102,
  NetworkTimeout = -400
}

// 手动创建枚举值的数组
export const HelloPlayErrorValues: HelloPlayError[] = [
  HelloPlayError.UnKnown,
  HelloPlayError.None,
  HelloPlayError.NoDataSource,
  HelloPlayError.NoStreamInDataSource,
  HelloPlayError.NoAudioVideoStream,
  HelloPlayError.NetworkTimeout
];